src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1846,5 +1846,32 @@ public static boolean isXnp(final String reference, final String alternate) {
 
         return result;
     }
+
+    /**
+     * Returns true if the context represents an MNP. If the context supplied contains the GVCF NON_REF symbolic
+     * allele, then the determination is performed after discarding the NON_REF symbolic allele.
+     *
+     * @param ctx a Variant context
+     * @return true if the context represents an unmixed MNP (i.e. all alleles are non-symbolic and of the same
+     *         length > 1), false otherwise
+     */
+    public static boolean isMnpWithoutNonRef(final VariantContext ctx) {
+        final List<Allele> alleles = ctx.getAlleles();
+        int length = -1;
```

Comment:
This code could be more concise as follows:

```
final int refAlleleLength = allleles.get(0).length();
if (alleles.stream().anyMatch(a -> a.length() < 2 || a.length() != refAlleleLength)) {
   return false;
} else {
   return alleles.stream().noneMatch(a -> a.isSymbolic() && a.equals(Allele.NON_REF_ALLELE));
}
```
Usually the decision to `return false` will be made on the first pass, so that's not really a performance issue, although the streams will be a lot slower than the `for` loop.  You can decide if that's a problem.

Reply:
I'd forgotten that the ref allele cannot be symbolic, thanks.  I've used that to make the method more concise, but still using the forloop:

```java
    public static boolean isUnmixedMnpIgnoringNonRef(final VariantContext vc) {
        final List<Allele> alleles = vc.getAlleles();
        int length = vc.getReference().length(); // ref allele cannot be symbolic
        if (length < 2) { return false; }

        for (final Allele a : alleles) {
            if (a.isSymbolic() && !Allele.NON_REF_ALLELE.equals(a)) { return false; }
            else if (!a.isSymbolic() && a.length() != length) { return false; }
        }

        return true;
    }
```